bridge_adaptivity/module/tests/test_models.py

```diff
@@ -115,3 +119,68 @@ def test_policy_cls_property(self):
         self.assertTrue(gp_trials_count.policy_cls is TrialsCountGradingPolicy)
         self.assertTrue(gp_points_earned.policy_cls is PointsEarnedGradingPolicy)
         self.assertTrue(gp_full_credit.policy_cls is FullCreditOnCompleteGradingPolicy)
+
+
+@ddt
+class TestActivityModel(TestCase):
+    fixtures = ['gradingpolicy.json', 'engine.json']
+
+    @patch('module.tasks.sync_collection_engines.apply_async')
+    def setUp(self, mock_apply_async):
+        self.user = BridgeUser.objects.create_user(
+            username='test',
+            password='test',
+            email='test@me.com'
+        )
+        self.consumer = LtiProvider.objects.create(
+            consumer_name='name',
+            consumer_key='key',
+            consumer_secret='secret',
+        )
+        self.lti_user = LtiUser.objects.create(
+            user_id='some_user', course_id='some_course', email=self.user.email,
+            lti_consumer=self.consumer, bridge_user=self.user
+        )
+        self.client.login(username='test', password='test')
+        # collections
+        self.collection1 = Collection.objects.create(name='col1', owner=self.user)
+        # grading policies
+        self.trials_count = GradingPolicy.objects.get(name='trials_count')
+        self.points_earned = GradingPolicy.objects.get(name='points_earned')
+        self.engine = Engine.objects.create(engine='engine_mock')
+        self.test_cg = CollectionGroup.objects.create(
+            name='TestColGroup',
+            owner=self.user,
+            engine=self.engine,
+            grading_policy=self.points_earned
+        )
+        self.test_cg.collections.add(self.collection1)
+        self.sequence = Sequence.objects.create(
+            lti_user=self.lti_user, collection=self.collection1,
+            engine=self.engine, grading_policy=self.trials_count,
+        )
+
+    @unpack
+    @data({'stype': 'video', 'is_problem': False},
+          {'stype': 'html', 'is_problem': False},
+          {'stype': 'problem', 'is_problem': True},
+          )
+    def test_is_problem_property(self, stype, is_problem):
+        activity = Activity(name='test', collection=self.collection1, tags='test', atype='G', stype=stype)
+        self.assertEqual(activity.is_problem, is_problem)
+
+    @unpack
+    @data({'stype': 'video', 'is_problem': False},
+          {'stype': 'html', 'is_problem': False},
+          {'stype': 'problem', 'is_problem': True},
+          )
+    def test_create_sequence_item_for_activity(self, stype, is_problem):
+        activity = Activity.objects.create(
+            name='test', collection=self.collection1, tags='test', atype='G', stype=stype
+        )
+        sequence_item = SequenceItem.objects.create(
+            sequence=self.sequence,
+            activity=activity,
+        )
+        self.assertTrue(sequence_item.is_problem == activity.is_problem)
```

Comment:
What do we test in L185 if it is the same by the definition?

Reply:
here we test that sequence has the same value for property is_problem as well as activity.is_problem. 

Also, it checks that logic of save method of Sequence object is working correctly.

Reply:
got it. Thanks